e2e/tests/plugins/notebook/tags.e2e.spec.js

```diff
@@ -0,0 +1,123 @@
+        await Promise.all([
+            page.waitForNavigation(/*{ url: 'http://localhost:8080/#/browse/mine/b8a33dd9-34c1-4981-82c7-0b5f45301fde?tc.mode=fixed&tc.startBound=1655456181296&tc.endBound=1655457981296&tc.timeSystem=utc&view=notebook-vue&pageId=ede28c83-206a-45cb-bcb0-cf5ba889af51&sectionId=4f75da12-0fa1-45aa-9174-1f21cb710fc0' }*/),
+            page.locator('button:has-text("OK")').click()
+        ]);
+
+        // Click text=To start a new entry, click here or drag and drop any object
+        await page.locator('text=To start a new entry, click here or drag and drop any object').click();
+    }
+
+    async function createNotebookEntryAndTags(page) {
+        await createNotebookAndEntry(page);
+        // Click text=To start a new entry, click here or drag and drop any object
+        await page.locator('button:has-text("Add Tag")').click();
+
+        // Click [placeholder="Type to select tag"]
+        await page.locator('[placeholder="Type to select tag"]').click();
+        // Click text=Driving
+        await page.locator('text=Driving').click();
+
+        // Click button:has-text("Add Tag")
+        await page.locator('button:has-text("Add Tag")').click();
+        // Click [placeholder="Type to select tag"]
+        await page.locator('[placeholder="Type to select tag"]').click();
+        // Click text=Science
+        await page.locator('text=Science').click();
+    }
+
+    test('Can load tags', async ({ page }) => {
+        await createNotebookAndEntry(page);
+        // Click text=To start a new entry, click here or drag and drop any object
+        await page.locator('button:has-text("Add Tag")').click();
+
+        // Click [placeholder="Type to select tag"]
+        await page.locator('[placeholder="Type to select tag"]').click();
+
+        await expect(page.locator('.c-input--autocomplete__options')).toContainText("Science");
+        await expect(page.locator('.c-input--autocomplete__options')).toContainText("Drilling");
+        await expect(page.locator('.c-input--autocomplete__options')).toContainText("Driving");
+    });
+    test('Can add tags', async ({ page }) => {
+        await createNotebookEntryAndTags(page);
+
+        await expect(page.locator('.c-notebook__entry')).toContainText("Science");
+        await expect(page.locator('.c-notebook__entry')).toContainText("Driving");
+
+        // Click button:has-text("Add Tag")
+        await page.locator('button:has-text("Add Tag")').click();
+        // Click [placeholder="Type to select tag"]
+        await page.locator('[placeholder="Type to select tag"]').click();
+
+        await expect(page.locator('.c-input--autocomplete__options')).not.toContainText("Science");
+        await expect(page.locator('.c-input--autocomplete__options')).not.toContainText("Driving");
+        await expect(page.locator('.c-input--autocomplete__options')).toContainText("Drilling");
+    });
+    test('Can search for tags', async ({ page }) => {
```

Comment:
@scottbell unfortunately, playwright has made the trade off for stability over speed in this respect: each browser launch has a totally separate context which prevents any test pollution or cross-talk for running tests... but means that the majority of a test execution time is getting to a loaded page. It can take up to 5 seconds in some cases. FWIW this avoids all of the issues we're experiencing in the unit tests.

Historically, the reason for smaller and more granular tests was to "increase the signal to noise ratio" in failed test reports. If 1 of these 3 test scenarios failed, someone would have an easier triage than if all of them failed.

With the video, images, and tracefiles coupled with our historical test reporting in circle ci it's incredibly easy to zero in on failure.

The only problem is with "reporting" each of these test scenarios. It's not entirely clear what all we're covering within a larger testcase unless we're overly verbose in the title. I _think_ the https://playwright.dev/docs/api/class-test#test-step API would allow us to report on these scenarios with more specificity but I haven't had time to play around with it